### PR TITLE
DE44690 - Account for fullPage mode when running pre/post-processing on HTML

### PR DESCRIPTION
**Pull request by grant-cleary**
When the `fullPage` plugin is enabled, our pre-processing and post-processing can end up wiping out the document head, so... Don't do that. 😅

### CODE REVIEW

**Review by dmd2l — APPROVED** [review]

**Review by PaulKrznaric — APPROVED** [review]

File: htmleditor-mixin.js [hunk 1]
```diff
@@ -285,12 +285,22 @@ export const HtmlEditorMixin = superclass => class extends Localizer(FormElement
...
 		converters.forEach(convert => convert(tempDoc));
-		return tempDoc.body.innerHTML;
+		if (!this.fullPage || sourceView) return tempDoc.body.innerHTML;
+
+		const docType = tempDoc.doctype;
+		if (!docType) return tempDoc.documentElement.outerHTML;
+
+		let docTypeId;
+		if (docType.publicId) docTypeId = ` PUBLIC "${docType.publicId}"`;
+		else if (docType.systemId) docTypeId = ' SYSTEM';
```
Comment (njostonehouse) [inline htmleditor-mixin.js:300]: Should this append `docType.systemId`?

File: htmleditor-mixin.js [hunk 2]
```diff
@@ -285,12 +285,22 @@ export const HtmlEditorMixin = superclass => class extends Localizer(FormElement
...
+
+		const docTypeString = `<!DOCTYPE ${docType.name}${docTypeId || ''}>`;
```
Comment (njostonehouse) [inline htmleditor-mixin.js:302]: Can `docType.name` be `null`?

File: htmleditor-mixin.js [hunk 3]
```diff
@@ -502,13 +512,13 @@ export const HtmlEditorMixin = superclass => class extends Localizer(FormElement
 						converters.push(htmlDoc => encodeSrcReplaceStrings(htmlDoc, this._context.orgUnitId));
 					}
 
-					e.content = this._convertHtml(e.content, converters);
+					e.content = this._convertHtml(e.content, e.source_view, converters);
 				});
 
-				editor.on('PostProcess', (e) => {
+				editor.on('GetContent', (e) => {
```
Comment (grant-cleary) [inline htmleditor-mixin.js:518]: `PostProcess` was earlier in the flow than it needed to be for this and also didn't contain the `source_view` information we needed.

File: htmleditor-mixin.js (hunk 1 above)
Comment (grant-cleary) [reply htmleditor-mixin.js:300]: Yes, it should. Good catch.

File: htmleditor-mixin.js (hunk 2 above)
Comment (grant-cleary) [reply htmleditor-mixin.js:302]: I don't think it can be if `docType` is defined, but better safe than sorry.

File: htmleditor-mixin.js [hunk 4]
```diff
@@ -285,12 +285,22 @@ export const HtmlEditorMixin = superclass => class extends Localizer(FormElement
...
 		converters.forEach(convert => convert(tempDoc));
-		return tempDoc.body.innerHTML;
+		if (!this.fullPage || sourceView) return tempDoc.body.innerHTML;
+
+		const docType = tempDoc.doctype;
+		if (!docType) return tempDoc.documentElement.outerHTML;
+
+		let docTypeId;
+		if (docType.publicId) docTypeId = ` PUBLIC "${docType.publicId}"`;
+		if (docType.systemId) docTypeId += ` ${docType.publicId ? '' : 'SYSTEM '}"${docType.systemId}"`;
```
Comment (grant-cleary) [inline htmleditor-mixin.js:300]: Ah crap, I need to init `docTypeId` for this to work. One sec.

**Review by njostonehouse — APPROVED** [review]

**Pull request merged**